### PR TITLE
reproduce decode interface map bug #299

### PR DESCRIPTION
**Pull request by alpha-baby**
#299

**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: decode_test.go [hunk 1]
```diff
@@ -223,7 +223,7 @@ func TestIssue299HessianDecode(t *testing.T) {
 			"interface_slice": []interface{}{
 				"MyName",
 			},
-			"interface_map": map[string]interface{}{
```
Comment (alpha-baby) [inline decode_test.go]: unsuport specific key types ??
I have tried Java and it is supported this case

Comment (wongoo) [reply decode_test.go]: currently hessian2 default encode as untyped map, see https://github.com/apache/dubbo-go-hessian2/blob/master/map.go#L122

Comment (wongoo) [reply decode_test.go]: @alpha-baby default encoding as untyped map does not match the hessian2 protocol, u are welcome to fix it if u will

Comment (alpha-baby) [reply decode_test.go]: > @alpha-baby default encoding as untyped map does not match the hessian2 protocol, u are welcome to fix it if u will

ok, thx

**Review by wongoo — APPROVED** [review]

**Review by zouyx — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**